packages/babel-parser/src/parser/expression.ts

```diff
@@ -1928,9 +1935,10 @@ export default abstract class ExpressionParser extends LValParser {
   }
 
   parseNewCallee(this: Parser, node: Undone<N.NewExpression>): void {
-    node.callee = this.parseNoCallExpr();
-    if (node.callee.type === "Import") {
-      this.raise(Errors.ImportCallNotNewExpression, { at: node.callee });
+    const callee = this.parseNoCallExpr();
+    node.callee = callee;
+    if (callee.type === "Import" || callee.type === "ImportExpression") {
```

Comment:
This is valid syntax, do we parse it correctly? `new (import(a))`

Reply:
Good catch, currently it does not work with the import expression AST. I will add a test and fix it.